### PR TITLE
feat: Adding new program files.

### PR DESCRIPTION
**Pull request by HKilaji**
-Frame.h is added. Contains class Frame, contains member function for size of board, contains draw function to draw frame around the board.

-main.cpp is added. Contains object for class Frame, and a draw function inside a loop.

### CODE REVIEW

File: Frame.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include<stdlib.h>
+
+#include<iostream>
+
+using namespace std;
+
+bool gameover = 0;
+
+class Frame{
+	public:
```
Comment (johansaji) [inline Frame.h:10]: All the data members should be private.

File: Frame.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	const int height = 50;
+	const int width = 50;
```
Comment (johansaji) [inline Frame.h:12]: The goal is that the frame the height and width to be configurable

File: Frame.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	int score = 0;
+		
+	void draw(){
```
Comment (johansaji) [inline Frame.h:15]: implementation should be moved to corresponding .cpp file

File: Frame.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+			printf("#");
+		    }
+		}
+		printf("\n");
+ 	     }
+	     for (int itr = 0; itr < width + 2; itr++){
+		printf("#");
+	     }
+	     printf("\n");
+	     printf("Score: %d \n", score);
```
Comment (johansaji) [inline Frame.h:37]: How are you going to render other elements in the game?

File: Frame.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
+#include<stdlib.h>
+
+#include<iostream>
+
+using namespace std;
+
+bool gameover = 0;
```
Comment (johansaji) [inline Frame.h:7]: Why is this variable declared here?

File: main.cpp [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#include<iostream>
+
+#include "Frame.h"
+
+using namespace std;
+int main(){
```
Comment (johansaji) [inline main.cpp:6]: add new line after using namespace

File: Frame.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#include<stdlib.h>
+
```
Comment (johansaji) [inline Frame.h:2]: no need to a new line in between

File: Frame.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+class Frame{
+	public:
+	const int height = 50;
+	const int width = 50;
+	int score = 0;
+		
+	void draw(){
+	     system("clear");
+	     for (int itr = 0; itr < width + 2; itr++){
```
Comment (johansaji) [inline Frame.h:17]: why width + 2?

File: main.cpp [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#include<iostream>
+
```
Comment (johansaji) [inline main.cpp:2]: no need for new line

**Review by johansaji — CHANGES_REQUESTED** [review]

**Comment by johansaji** [comment]
Not Compiling
```ZONDEV\jsaji:snake-cpp$ gcc -Wall main.cpp
In file included from main.cpp:3:0:
Frame.h:11:21: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11
  const int height = 50;
                     ^
Frame.h:12:20: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11
  const int width = 50;
                    ^
Frame.h:13:14: warning: non-static data member initializers only available with -std=c++11 or -std=gnu++11
  int score = 0;
              ^
Frame.h: In member function ‘void Frame::draw()’:
Frame.h:18:13: error: ‘printf’ was not declared in this scope
   printf("#");
             ^
Frame.h:20:18: error: ‘printf’ was not declared in this scope
       printf("\n");
                  ^
ZONDEV\jsaji:snake-cpp$
```

File: Frame.h (hunk 1 above)
Comment (HKilaji) [reply Frame.h:10]: I will fix those in the next update.

File: Frame.h (hunk 2 above)
Comment (HKilaji) [reply Frame.h:12]: I know, right now I am trying to create the structure of game screen. Once succeed I will change it to be configurable.

File: Frame.h (hunk 3 above)
Comment (HKilaji) [reply Frame.h:15]: Yes, I will do that currently not structure of code is not complete. Once done I will put the in their corresponding files.

File: Frame.h (hunk 4 above)
Comment (HKilaji) [reply Frame.h:37]: I will start the coding for rendering after I finish initial setup of game.

File: Frame.h (hunk 5 above)
Comment (HKilaji) [reply Frame.h:7]: currently This variable has no suitable class but is important thats why it is declared there. In the next step of code I will declare it in proper place.

File: main.cpp (hunk 6 above)
Comment (HKilaji) [reply main.cpp:6]: okay I will fix this in next update.

File: Frame.h (hunk 7 above)
Comment (HKilaji) [reply Frame.h:2]: okay I will fix this in next update.

File: Frame.h (hunk 8 above)
Comment (HKilaji) [reply Frame.h:17]: I will fix that in next step. I am trying to find a nice way to print the frame. 

I did width + 2 to complete the frame as there is two corner walls are missing , but didn't find right way that time.

File: main.cpp (hunk 9 above)
Comment (HKilaji) [reply main.cpp:2]: okay I will fix this in next update.

**Comment by HKilaji** [comment]
I will fix the compilation problem in the next update.

**Pull request closed without merging**